### DISCUSSION TITLE
Question / suggestion about charging points

### DISCUSSION THREAD

**Discussion by OpenGreenStreet** [streetcomplete/StreetComplete#3445] [Ideas]
Does StreetComplete query all information relevant for e-cars and e-bikes (plug type, charging power, costs, payment method, etc.) and does it do so at sensible time intervals? (see also: https://de.wikipedia.org/wiki/Diskussion:Ladepunkt)
Would it be possible to store photos (permanently (!)) and enter fault messages in the future?

**Comment by FloEdelmann** [comment]
* Plug type: Please see #903 and add any relevant information there to help implement this quest.
* Capacity: This is already asked: https://github.com/streetcomplete/StreetComplete/blob/v36.1/app/src/main/java/de/westnordost/streetcomplete/quests/charging_station_capacity/AddChargingStationCapacity.kt
* Operator: This is already asked: https://github.com/streetcomplete/StreetComplete/blob/v36.1/app/src/main/java/de/westnordost/streetcomplete/quests/charging_station_operator/AddChargingStationOperator.kt

Please also see closed issues: https://github.com/streetcomplete/StreetComplete/issues?q=is%3Aissue+charging+station+is%3Aclosed+

**Reply by OpenGreenStreet** [reply]
@FloEdelmann: Thank you for your feedback: Do I understand you correctly? All data are available. They just need to be displayed correctly by www.flosm.de?
( https://www.flosm.de/html/POI-Karte.html?sw=chargingbike,chargingcar,chargingscooter,chargingstation,chargingtruck,chargingwheelchair )

**Reply by matkoniecz** [reply]
>  All data are available. They just need to be displayed correctly by www.flosm.de?

It is posible that something is not yet surveyed

**Reply by OpenGreenStreet** [reply]
Yes, it depends on us mappers, but all necessary fields are queried?

**Reply by FloEdelmann** [reply]
@OpenGreenStreet Please note that StreetComplete is not the same as OpenStreetMap! StreetComplete shows quests to users that in turn add data to an already-existing element in OSM. E.g. the `AddChargingStationCapacity` quests searches for all nodes that are tagged with `amenity=charging_station` but lack the `capacity` tag.

In OSM, there are many mapped charging stations. But keep in mind that it is a community-driven project, so some of those will be outdated or incomplete, many others will be missing. But the tags that should be used to record all the information you asked for are all well-defined: https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dcharging_station

How other websites / apps / projects (like the FLOSM map you linked) make use of those data is a whole different story. The complaint was that this particular map displays outdated information. Maybe their copy of the OSM database is old, maybe the data in OSM for some place were indeed not updated to reflect the current situation. The few charging stations that I know are included and up-to-date, at least.

There is another map that I know of that displays the charging stations in OSM, but that one is more focused on recording new information rather than showing the information currently present: https://mapcomplete.osm.be/charging_stations.html

All in all, I don't understand exactly what your question regarding charging station support in StreetComplete is about.

**Reply by OpenGreenStreet** [reply]
Thanks for the detailed feedback. I contribute to OSM almost exclusively via StreetComplete. My point here is to understand if the criticism expressed here (https://de.wikipedia.org/wiki/Diskussion:Ladepunkt)  is basically solved in OSM and if StreetComplete queries all data.
If I understand it correctly, OSM provides - as far as they are recorded - all data and StreetComplete also queries all data. The display of www.flosm.de would have to be optimized!?

**Reply by FloEdelmann** [reply]
> OSM provides - as far as they are recorded - all data

Yes.

> StreetComplete also queries all data

No, just where it makes sense as a StreetComplete quest. As I said, capacity and operator can be recorded with StreetComplete already. For plug types there is an open issue (#903), so it is not yet implemented; more work is needed there. Others were already declined (https://github.com/streetcomplete/StreetComplete/issues?q=is%3Aissue+charging+station+is%3Aclosed+sort%3Aupdated-desc), others were not yet suggested.

> The display of [www.flosm.de](http://www.flosm.de) would have to be optimized!?

Maybe, I don't know that project. But they already show all information as recorded in OSM, e.g. `socket:type2 = 1`. Maybe they want it like that, maybe they want to display images of the plugs instead/additionally. Maybe they need to update their copy of the OSM database more often, maybe that is too much effort. I can't say.

**Reply by OpenGreenStreet** [reply]
thank you very much!
